chains/evm/cli/admin/pause.go

```diff
@@ -14,45 +22,71 @@ var pauseCmd = &cobra.Command{
 	PreRun: func(cmd *cobra.Command, args []string) {
 		logger.LoggerMetadata(cmd.Name(), cmd.Flags())
 	},
-	Run: pause,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return PauseCmd(cmd, args, evmtransaction.NewTransaction, &evmgaspricer.LondonGasPriceDeterminant{})
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		err := ValidatePauseCmdFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		ProcessPauseCmdFlags(cmd, args)
+
+		return nil
+	},
 }
 
-func BindPauseFlags(cmd *cobra.Command) {
+func BindPauseCmdFlags(cmd *cobra.Command) {
 	cmd.Flags().StringVar(&Bridge, "bridge", "", "bridge contract address")
 	flags.MarkFlagsAsRequired(cmd, "bridge")
 }
+
 func init() {
-	BindPauseFlags(pauseCmd)
+	BindPauseCmdFlags(pauseCmd)
 }
 
-func pause(cmd *cobra.Command, args []string) {
-	log.Debug().Msgf(`
-Pausing
-Bridge address: %s`, Bridge)
+func ValidatePauseCmdFlags(cmd *cobra.Command, args []string) error {
+	if !common.IsHexAddress(Bridge) {
+		return fmt.Errorf("invalid bridge address: %s", Bridge)
+	}
+	return nil
 }
 
-/*
-func pause(cctx *cli.Context) error {
-	url := cctx.String("url")
-	gasLimit := cctx.Uint64("gasLimit")
-	gasPrice := cctx.Uint64("gasPrice")
-	sender, err := cliutils.DefineSender(cctx)
-	if err != nil {
-		return err
-	}
-	bridgeAddress, err := cliutils.DefineBridgeAddress(cctx)
+func ProcessPauseCmdFlags(cmd *cobra.Command, args []string) {
+	bridgeAddr = common.HexToAddress(Bridge)
+}
+
+func PauseCmd(cmd *cobra.Command, args []string, txFabric calls.TxFabric, gasPricer utils.GasPricerWithPostConfig) error {
+	// fetch global flag values
+	url, gasLimit, gasPrice, senderKeyPair, err := flags.GlobalFlagValues(cmd)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not get global flags: %v", err)
 	}
-	ethClient, err := client.NewClient(url, false, sender, big.NewInt(0).SetUint64(gasLimit), big.NewInt(0).SetUint64(gasPrice), big.NewFloat(1))
+
+	fmt.Printf("Pausing bridge: %s\n", Bridge)
```

Comment:
This is more of an discussion.
We used logger at INFO level to log CMD messages to user and should probably 
discuss if we want to use some pretty format in the future, but for now I would keep it consistent.

Reply:
agreed